modules/profile/manifests/st2server.pp

```diff
@@ -689,71 +689,20 @@
     ],
   }
 
-  # ### Nginx needs access to make calls to PAM, and by
-  # ### extension, needs access to /etc/shadow to validate users.
-  # ### Let's at least try to do this safely and consistently
-
-  $_st2auth_custom_options = 'limit_except OPTIONS {
-    auth_pam "Restricted";
-    auth_pam_service_name "nginx";
-    }'
-
-
-  # ## Authentication
-  # Note: We need to return a custom 401 error since nginx pam module intercepts
-  # 401 and there is no other way to do it :/
-  $_st2auth_custom_401_error_handler = '
-  error_page 401 =401 @401_response;
-
-  location @401_response {
-    more_set_headers "Access-Control-Allow-Origin: *";
-    more_set_headers "Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS";
-    more_set_headers "Access-Control-Allow-Credentials: true";
-    return 401 "Invalid or missing credentials";
-  }'
-
   # Note 1: We don't need an if block since more_set_headers only sets header if
   # already set so duplicate headers are ot a problem.
   # Note 2: This module requires nginx-extras to be installed.
   # Note 3: We use MoreSetHeaders module since old version of nginx we use
   # doesn't support overriding / setting headers on non-succesful responses.
+
+  # NOTE: @kami: here is the issue. more_set_headers is not allowed on RHEL, because
```

Comment:
I'll have a look, now that we don't use pam module anymore it might just work (previously nginx was not propagating upstreams headers on pam auth failure).